### PR TITLE
feat: integrate Privacy

### LINKED ISSUE #124: Integrate privacy.js
Based on https://github.com/adobe/fedpub/blob/blocks/hub/scripts/scripts.js#L278-L302

### PR DESCRIPTION
**Pull request by narcis-radu**
## Description
Integrate Privacy.js and add support for GDPR. Fixes #124. **The banner will be visible to all users coming from GDPR enforced countries (EU + CA).**
**NOTE**: the banner will always pop-up on localhost, hlx3.page, hlx.live, even if consent has been provided, since these domains don't have a OneTrust id (yet). I will ask the team to create IDs for hlx3.page and hlx.live

**NOTE ON PERFORMANCE**
privacy.min.js loads OTSDKStub.js (external) which loads:
* configuration for domain - e.g. https://cdn.cookielaw.org/consent/7a5eb705-95ed-4cc4-a11d-0cc5760e93db/7a5eb705-95ed-4cc4-a11d-0cc5760e93db.json
* banner file - https://cdn.cookielaw.org/scripttemplates/6.9.0/otBannerSdk.js
* network call to get location - https://geolocation.onetrust.com/cookieconsentpub/v1/geo/location
* localized content - https://cdn.cookielaw.org/consent/7a5eb705-95ed-4cc4-a11d-0cc5760e93db/8fc5213e-cec6-4fca-a134-aec9029b0675/en.json
* banner css - https://cdn.cookielaw.org/scripttemplates/6.9.0/assets/otFlat.json
* PC css - https://cdn.cookielaw.org/scripttemplates/6.9.0/assets/otPcCenter.json

The files are loaded every single time by the OneTrust script.

URL for testing:
- https://privacy--business-website--adobe.hlx3.page/blog/

## Motivation and Context
Adds support for GDPR

## How Has This Been Tested?
Locally

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/delayed.js [hunk 1]
```diff
@@ -51,3 +51,37 @@ if (window.hlx.rum.isSelected) {
...
+  const preferenceCenter = document.querySelector('[href="https://www.adobe.com/#openPrivacy"]');
+  if (preferenceCenter) {
+    preferenceCenter.addEventListener('click', (event) => {
+      event.preventDefault();
+      window?.adobePrivacy?.showConsentPopup();
+    });
+  }
+
+  const env = getHelixEnv().name === 'prod' ? '' : 'stage.';
+  loadScript(`https://www.${env}adobe.com/etc/beagle/public/globalnav/adobe-privacy/latest/privacy.min.js`);
```
Comment (narcis-radu) [inline scripts/delayed.js:84]: privacy.min.js loads OTSDKStub.js which loads:
* configuration for domain - e.g. https://cdn.cookielaw.org/consent/7a5eb705-95ed-4cc4-a11d-0cc5760e93db/7a5eb705-95ed-4cc4-a11d-0cc5760e93db.json
* banner file - https://cdn.cookielaw.org/scripttemplates/6.9.0/otBannerSdk.js
* network call to get location - https://geolocation.onetrust.com/cookieconsentpub/v1/geo/location
* localized content - https://cdn.cookielaw.org/consent/7a5eb705-95ed-4cc4-a11d-0cc5760e93db/8fc5213e-cec6-4fca-a134-aec9029b0675/en.json
* banner css - https://cdn.cookielaw.org/scripttemplates/6.9.0/assets/otFlat.json
* PC css - https://cdn.cookielaw.org/scripttemplates/6.9.0/assets/otPcCenter.json

The files are loaded every single time by the OneTrust script.

**Comment by davidnuescheler** [comment]
i am not sure i have enough context to review this, i am happy to resolve the merge conflicts if that helps though. thanks so much to @narcis-radu for the PR.

**Comment by narcis-radu** [comment]
Hi @davidnuescheler - the "cookie preferences" button in the footer does not allow users to change their tracking preferences (see #127). This is because the PrivacyJS integration is currently missing from the site and we don't fully respect GDPR regulations. I noticed #124 requested an implementation of PrivacyJS and I went ahead and did it via this PR. The issue that I see is with LH scores and this is because how OneTrust loads their scripts (which we have no control over). Everything was added in "delayed", but I guess it's not good enough.
cc @auniverseaway

**Comment by kptdobe** [comment]
@davidnuescheler If you load https://privacy--business-website--adobe.hlx3.page/blog/ in incognito mode, you'll se the "Make it Your Own" bar at the bottom which appears. I assume this have an impact on the LH. I wanted to get your feedback on that.

**Comment by narcis-radu** [comment]
I added a small comment in description - the banner should only be visible to users coming from EU + CA (where GDPR is mandatory). Regardless of this, the banner assets are currently loaded every single time by the OneTrust code.

**Review by auniverseaway — APPROVED** [review]

**Comment by auniverseaway** [comment]
@narcis-radu Just need to fix the merge conflicts and we should be good. Thanks for doing this!

**Review by davidnuescheler — APPROVED** [review]

**Pull request merged**